### PR TITLE
Adds optional scale and bias to cudnn's layernorm

### PR DESCRIPTION
**Pull request by vedaanta**
<details>
  <summary><b>Before submitting</b></summary>

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/main/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

</details>

## What does this PR do?
This PR adds more functionality to cudnn's layernorm
1. Adds optional bias and scale
2. Updates to cudnn 9.1 requirements
3. Removes older way of caching full graph making function

Future:
1. Add benchmarks for LN fwd with litgpt configs
2. Add LN bwd
3. Add RMSnorm 

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

### CODE REVIEW

File: thunder/executors/cudnn_layernormex.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
-from typing import Any
+from typing import Any, List
```
Comment (crcrpar) [inline thunder/executors/cudnn_layernormex.py:1]: doesn't seem to be used
```suggestion
from typing import Any
```

File: thunder/executors/cudnn_layernormex.py [hunk 2]
```diff
@@ -18,117 +19,174 @@
...
 
     a_4d, weight_4d, bias_4d = _transform_layer_norm_inputs(a, normalized_shape, weight, bias)
 
     try:
         _make_cudnn_layer_norm_graph(a_4d, weight_4d, bias_4d)
-    except:
+    except cudnn.cudnnGraphNotSupportedError as ex:
         return False
+    except Exception as e:
+        raise
```
Comment (crcrpar) [inline thunder/executors/cudnn_layernormex.py:182]: when would we hit this? I'm not sure if raise is appropriate here at glance

File: thunder/executors/cudnn_layernormex.py (hunk 1 above)
Comment (vedaanta-nvidia) [reply thunder/executors/cudnn_layernormex.py:1]: Seems like pre-commit changed the type hint from `List` to `list`.
[[Here](https://github.com/Lightning-AI/lightning-thunder/pull/234/commits/0250461e8372544a526f2983dd67972fbcf648f1#diff-0729becfc13c06019a75178ac7afe4cebf0382a709b88fd932ac65fc5644a181R167)]

**Comment by lantiga** [comment]
tagging older draft PRs as `later`, feel free to reopen if this gets back being active

**Pull request closed without merging**